recipes/mawk/all/test_package/conanfile.py

```diff
@@ -0,0 +1,15 @@
+from conan import ConanFile
+from conan.tools.build import can_run
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "VirtualBuildEnv"
+    test_type = "explicit"
+
+    def build_requirements(self):
+        self.tool_requires(self.tested_reference_str)
+
+    def test(self):
+        if can_run(self):
+            self.run("mawk --version")
```

Comment:
The test package should simple enough, we need to check if the package is fine, not checking project features.

Reply:
the problem with `--version` is that it doesn't tell you definitively if you've packaged it correctly. We've seen in the past that a number of applications bypass functional elements of their code base to provide version information, which means that when you run the test everything looks fine, but when you go to use the application it becomes obvious that it has been packaged incorrectly or dependencies are insufficiently modelled. [This was the whole problem with the `gfortran` recipe](https://github.com/conan-io/conan-center-index/issues/14120), which used `--version` to assess whether it was packaged correctly. The test_package indicated it was, but when attempting to use the application it would error out.

Without testing a functional part of the application, you would need to know how it's implemented to know if `--version` is an appropriate check. I'd suggest determining this is too onerous, so it makes more sense to do a simple functional check. What I've proposed above is probably the simplest it could be for `awk`.

Also, `mawk --version` is invalid syntax from my experimentation on ubuntu 18.04.